conda/recipes/cuml-cpu/meta.yaml

```diff
@@ -33,8 +33,9 @@ requirements:
     - python x.x
     - numpy>=1.23,<3.0a0
     - pandas
-    - scikit-learn=1.2
-    - hdbscan>=0.8.38,<0.8.39
+    - packaging
+    - scikit-learn==1.5.*
+    - hdbscan>=0.8.39,<0.8.40
```

Comment:
These changes make `cuml-cpu` match the versions `cuml` is tested against.

https://github.com/rapidsai/cuml/blob/fbcaf123d582e349fd7a111fea75212bc8d5d8e3/dependencies.yaml#L441

https://github.com/rapidsai/cuml/blob/fbcaf123d582e349fd7a111fea75212bc8d5d8e3/dependencies.yaml#L484

And might help with #6340